e2e/zombienet/src/hooks/useContractQuery.test.ts

```diff
@@ -39,6 +39,62 @@ describe('useContractQuery', () => {
     });
   });
 
+  it('should automatically call refresh on new block when watch is enabled', async () => {
+    const { result: rawContract } = renderHook(() => useRawContract<Psp22ContractApi>(psp22Metadata, contractAddress), {
+      wrapper,
+    });
+
+    await waitFor(() => {
+      expect(rawContract.current.contract).toBeDefined();
+      expect(rawContract.current.contract?.client.options.signer).toBeDefined();
+    });
+
+    const contract = rawContract.current.contract;
+
+    const { result: balanceOf } = renderHook(
+      () => useContractQuery({ contract, fn: 'psp22BalanceOf', args: [ALICE], watch: true }),
+      {
+        wrapper,
+      },
+    );
+
+    await waitFor(() => {
+      expect(balanceOf.current.data).toBeDefined();
+    });
+
+    const beforeTranfer = balanceOf.current.data;
+
+    console.log('Before transfer:', beforeTranfer);
+
+    const { result: transfer } = renderHook(
+      () => useContractTx(contract, 'psp22Transfer'), // prettier-end-here
+      {
+        wrapper,
+      },
+    );
+
+    await new Promise<void>((resolve) => {
+      transfer.current.signAndSend({
+        args: [BOB, BigInt(1e12), '0x'],
+        callback: ({ status }) => {
+          if (status.type === 'Finalized') {
+            resolve();
+          }
+        },
+      });
+    });
+    console.log('Transfer completed!');
+
+    await waitFor(
+      () => {
+        expect(balanceOf.current.data).not.toEqual(beforeTranfer);
```

Comment:
@1cedrus  What the exact balance after the transfer? We should add that assertion here as well, not just to check the balance is different.